rocks/RocksIterator.java

```diff
@@ -75,40 +76,55 @@ public final boolean hasNext() {
                 return true;
             case EMPTY:
                 return fetchAndCheck();
+            case SEEKED_EMPTY:
+                return checkValidNext();
             case INIT:
                 return initialiseAndCheck();
             default: // This should never be reached
                 return false;
         }
     }
 
-    private synchronized boolean initialiseAndCheck() {
+    @Override
+    public void seek(T target) {
+        if (state == State.INIT) initialise(target.getBytes());
+        else internalRocksIterator.seek(target.getBytes().getArray());
+        state = State.SEEKED_EMPTY;
+    }
```

Comment:
because we seek but don't want to immediately check if there's a valid next, we introduce a new state `SEEKED_EMPTY` which will just check the current key without calling `next()` (which progresses the rocksDB iterator)